agent/agent.go

```diff
@@ -176,8 +176,7 @@ func (a *Agent) runTask(ctx context.Context, tc *taskContext) error {
 	complete := make(chan string)
 	go a.startTask(innerCtx, tc, complete)
 
-	status := a.wait(innerCtx, tc, heartbeat, complete)
-
+	status := a.wait(ctx, tc, heartbeat, complete)
```

Comment:
This is nearly the right thing to do, but `wait` relies on the inner context cancelation. I suppose we could ensure that `startTask` always sends on the channel. Otherwise `wait` will block indefinitely.

Also, we should write a test (include it in `CommandTestSuite/TestTimeout`?) that covers this behavior.

Reply:
I pushed (and rebased, sorry for the noise) passing wait both contexts, which is weird (and outlines that we are giving it three things named context)